### ISSUE TITLE
object 'repo' not found

### ISSUE THREAD

**Issue by hinkelman** [ficonsulting/RInno#88]
I'm trying to create my first RInno installer. When I run this code (sorry, not a reproducible example):

```
create_app(
  app_name  = "DSM2 Vis",
  app_dir   = "../dsm2-hydro-visualization-rinno/",
  include_R = FALSE,
  pkgs = c("shiny", "shinyWidgets", "leaflet", "dplyr", "ggplot2", "lubridate", "shinyjs", "MESS", "rgdal", "plyr", "Cairo", "cairoDevice"))
```

I get this error:

> Error in eval(parse(text = code, keep.source = FALSE), envir) : object 'repo' not found

I'm not really sure what is happening here. The problem might be in create_config(), but I came up empty in my flailing attempts. 

I don't get the same error when I run

```
example_app(wd = getwd())
create_app(app_name = "Your appname", app_dir = "app")
```
But in that case when I run compile_iss(), I get the following error

> Line 21: Resource update error: EndUpdateResource failed (5)

I'm using the latest GitHub version of the package; not the CRAN version.

Thanks for all your work on this package!

**Comment by hinkelman** [comment]
I sort of accidentally discovered that if you run the same call to create_app twice in a row, then it will run correctly on the 2nd call. I was then able to get a working installer (but I've only tested it on the same machine where the app was built; not one without R installed). Thanks again for creating such a great package!

**Comment by Dripdrop12** [comment]
I wasn't able to reproduce this, but it might have been just the specific commit that you were running. Re-open it if you see it again!

**Comment by Hopellopo** [comment]
Hey there! Sorry to resurrect this thread, but I am running into the same error. Here is a minimum reproducible example using the histogram app that comes with shiny:

[testApp.zip](https://github.com/ficonsulting/RInno/files/2294043/testApp.zip)

Upon entering the folder, I run: 
create_app("testApp")

This results in the error:
Error in eval(parse(text = text, keep.source = FALSE), envir) : 
  object 'repo' not found

I am on the latest github version of RInno, and R 3.5.1. Windows 7. Notably, this error does not come up for me on the CRAN version of RInno, however that version has another bug which renders it unusable.

Thanks for the package, though! It is super helpful.

**Comment by Dripdrop12** [comment]
@Hopellopo 

Can you try the following and let me what your output looks like?
```
devtools::install_github("ficonsulting/RInno")
packageVersion("RInno")
getwd()
```
`packageVersion` should return "0.3.0.9000" and `getwd` should be the location of app.R

Afterwards, re-run `create_app` and let me know how that works. I successfully installed testApp just now.

**Comment by Hopellopo** [comment]
> devtools::install_github("ficonsulting/RInno")
Downloading GitHub repo ficonsulting/RInno@master
from URL https://api.github.com/repos/ficonsulting/RInno/zipball/master
Installing RInno
"C:/PROGRA~1/R/R-35~1.1/bin/x64/R" --no-site-file --no-environ --no-save --no-restore --quiet CMD INSTALL  \
  "C:/Users/Hopellopo/AppData/Local/Temp/RtmpqWqoYY/devtools1c607486243b/ficonsulting-RInno-fdd2100" --library="C:/Program Files/R/R-3.5.1/library"  \
  --install-tests 

* installing *source* package 'RInno' ...
** R
** inst
** tests
** byte-compile and prepare package for lazy loading
** help
*** installing help indices
  converting help for package 'RInno'
    finding HTML links ... done
    add_pkgs                                html  
    code_section                            html  
    compile_iss                             html  
    copy_installation                       html  
    cran_version                            html  
    create_app                              html  
Rd warning: C:/Users/Hopellopo/AppData/Local/Temp/RtmpqWqoYY/devtools1c607486243b/ficonsulting-RInno-fdd2100/man/create_app.Rd:52: file link 'pandoc_version' in package 'rmarkdown' does not exist and so has been treated as a topic
    create_bat                              html  
    create_config                           html  
    directives_section                      html  
Rd warning: C:/Users/Hopellopo/AppData/Local/Temp/RtmpqWqoYY/devtools1c607486243b/ficonsulting-RInno-fdd2100/man/directives_section.Rd:23: file link 'pandoc_version' in package 'rmarkdown' does not exist and so has been treated as a topic
    download_packages                       html  
    example_app                             html  
    files_section                           html  
    flexdashboard_check                     html  
    get_Chrome                              html  
    get_Pandoc                              html  
Rd warning: C:/Users/Hopellopo/AppData/Local/Temp/RtmpqWqoYY/devtools1c607486243b/ficonsulting-RInno-fdd2100/man/get_Pandoc.Rd:12: file link 'pandoc_version' in package 'rmarkdown' does not exist and so has been treated as a topic
    get_R                                   html  
    get_Rtools                              html  
    icons_section                           html  
    install_inno                            html  
    languages_section                       html  
    run_section                             html  
    sanitize_R_version                      html  
    setup_section                           html  
    standardize_pkgs                        html  
    start_iss                               html  
    tasks_section                           html  
** building package indices
** installing vignettes
** testing if installed package can be loaded
*** arch - i386
*** arch - x64
* DONE (RInno)
In R CMD INSTALL
> packageVersion("RInno")
[1] ‘0.3.0.9000’
> getwd()
[1] "C:/Users/Hopellopo/Desktop/testApp"
> RInno::create_app("testApp", pkgs = c("shiny"))

Getting package dependencies... 

Downloading required packages...
trying URL 'http://cran.rstudio.com/bin/windows/contrib/3.5/httpuv_1.4.5.zip'
Content type 'application/zip' length 1182327 bytes (1.1 MB)
downloaded 1.1 MB

trying URL 'http://cran.rstudio.com/bin/windows/contrib/3.5/mime_0.5.zip'
Content type 'application/zip' length 46959 bytes (45 KB)
downloaded 45 KB

trying URL 'http://cran.rstudio.com/bin/windows/contrib/3.5/jsonlite_1.5.zip'
Content type 'application/zip' length 1203108 bytes (1.1 MB)
downloaded 1.1 MB

trying URL 'http://cran.rstudio.com/bin/windows/contrib/3.5/xtable_1.8-2.zip'
Content type 'application/zip' length 766146 bytes (748 KB)
downloaded 748 KB

trying URL 'http://cran.rstudio.com/bin/windows/contrib/3.5/digest_0.6.15.zip'
Content type 'application/zip' length 192879 bytes (188 KB)
downloaded 188 KB

trying URL 'http://cran.rstudio.com/bin/windows/contrib/3.5/htmltools_0.3.6.zip'
Content type 'application/zip' length 659965 bytes (644 KB)
downloaded 644 KB

trying URL 'http://cran.rstudio.com/bin/windows/contrib/3.5/R6_2.2.2.zip'
Content type 'application/zip' length 334778 bytes (326 KB)
downloaded 326 KB

trying URL 'http://cran.rstudio.com/bin/windows/contrib/3.5/sourcetools_0.1.7.zip'
Content type 'application/zip' length 530511 bytes (518 KB)
downloaded 518 KB

trying URL 'http://cran.rstudio.com/bin/windows/contrib/3.5/later_0.7.3.zip'
Content type 'application/zip' length 608133 bytes (593 KB)
downloaded 593 KB

trying URL 'http://cran.rstudio.com/bin/windows/contrib/3.5/promises_1.0.1.zip'
Content type 'application/zip' length 686934 bytes (670 KB)
downloaded 670 KB

trying URL 'http://cran.rstudio.com/bin/windows/contrib/3.5/crayon_1.3.4.zip'
Content type 'application/zip' length 749270 bytes (731 KB)
downloaded 731 KB

trying URL 'http://cran.rstudio.com/bin/windows/contrib/3.5/rlang_0.2.1.zip'
Content type 'application/zip' length 822667 bytes (803 KB)
downloaded 803 KB

trying URL 'http://cran.rstudio.com/bin/windows/contrib/3.5/Rcpp_0.12.18.zip'
Content type 'application/zip' length 4556971 bytes (4.3 MB)
downloaded 4.3 MB

trying URL 'http://cran.rstudio.com/bin/windows/contrib/3.5/BH_1.66.0-1.zip'
Content type 'application/zip' length 17880019 bytes (17.1 MB)
downloaded 17.1 MB

trying URL 'http://cran.rstudio.com/bin/windows/contrib/3.5/magrittr_1.5.zip'
Content type 'application/zip' length 155753 bytes (152 KB)
downloaded 152 KB

trying URL 'http://cran.rstudio.com/bin/windows/contrib/3.5/shiny_1.1.0.zip'
Content type 'application/zip' length 3635282 bytes (3.5 MB)
downloaded 3.5 MB

Error in eval(parse(text = text, keep.source = FALSE), envir) : 
  object 'repo' not found




Same error. Notably, after it made all of that noise about rmarkdown not existing, I manually installed it and tried again. it still threw  Error in eval(parse(text = text, keep.source = FALSE), envir) : 
  object 'repo' not found


Thanks for the prompt response, by the way!

**Comment by Dripdrop12** [comment]
It looks like that code is a `glue` transformer - https://github.com/tidyverse/glue/blob/master/R/transformer.R

I searched for all the places where 'repo' is used in RInno, and there shouldn't be an issue there (it is consistently defined). 

Maybe you have to update your package dependencies?

**Comment by Hopellopo** [comment]
So I called remove.packages() on RInno, devtools, glue, curl, httr, installr, jsonlite, magrittr, shiny, and stringr, before rebooting the R session. I reinstalled devtools before running:
devtools::install_github("ficonsulting/RInno")
packageVersion("RInno")
getwd()

and achieved expected results. I then ran create_app, and no dice. An identical error. I also tried uninstalling glue and reinstalling it from gihub rather than CRAN, and still no changes.
For what it is worth, I tried rolling back RInno, and found that commit d446df4f134713801766987a3de7bd626bb98ac9 works perfectly. I don't know if that is helpful for you.
